tests/common.go

```diff
@@ -52,13 +52,15 @@ func DeleteFirewall(firewalls *compute.FirewallsClient, gcpProject, firewallName
 		Project:  gcpProject,
 	}
 
-	// Explicitly do not wait for the deletion to complete. This makes the
-	// tests significantly slower
-	_, err := firewalls.Delete(context.Background(), req)
+	op, err := firewalls.Delete(context.Background(), req)
 	Expect(err).WithOffset(1).To(Or(
 		Not(HaveOccurred()),
 		BeGoogleAPIErrorWithStatus(http.StatusNotFound),
 	))
+
+	if op != nil {
+		Expect(op.Wait(context.Background())).WithOffset(1).To(Succeed())
```

Comment:
`Wait` will block until the `Operation` resource reaches the `Done` state. It does this by doing GET requests with exponential backoff. Initially I thought it stopped after the default one minute backoff period, but apparently it will continue polling in _regular_ intervals after the initial backoff. So this will either reach the `Done` state or it will hit an error. 

I guess in the test this is fine(-ish) but in the actual code we should set a timeout using the context.